src/coordination/coordinator_state_manager.cpp

```diff
@@ -73,29 +90,36 @@ auto CoordinatorStateManager::save_config(cluster_config const &config) -> void
   ptr<buffer> buf = config.serialize();
   cluster_config_ = cluster_config::deserialize(*buf);
   logger_.Log(nuraft_log_level::TRACE, "Saving cluster config to RocksDb");
-  auto const servers_vec =
-      ranges::views::transform(
-          cluster_config_->get_servers(),
-          [this](auto const &server) {
-            logger_.Log(nuraft_log_level::TRACE,
-                        fmt::format("Creating cluster config with id: {}, endpoint: {} and aux data: {} to disk.",
-                                    static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux()));
-            return std::tuple{static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux()};
-          }) |
-      ranges::to<std::vector>();
-  kv_store_.Put(kServersKey, nlohmann::json(servers_vec).dump());
+  auto json = SerializeClusterConfig(config);
```

Comment:
when you add to_json and from_json you should be able to call just nlohman::json(config)